zapisy/apps/grade/poll/views.py

```diff
@@ -182,6 +185,53 @@ def __get_counter_for_categories(polls):
 
         return number_of_submissions_for_category
 
+    @staticmethod
+    def __are_read(polls, user):
+        is_read_category = defaultdict(True.__bool__)
+        is_read_poll = dict()
+
+        last_views: Dict[Poll, datetime.datetime] = dict(
+                PollView.objects.filter(user=user, poll__in=polls).values_list("poll", "time")
+            )
+
+        last_modifieds: Dict[Poll, datetime.datetime] = dict(
+            Submission.objects.filter(poll__in=polls, submitted=True).order_by('poll',
+                    'modified').distinct('poll').values_list("poll", "modified")
+        )
+        for poll in polls:
+            if poll.id in last_views:
+                if poll.id in last_modifieds:
+                    is_read_poll[poll] = (last_views[poll.id] > last_modifieds[poll.id])
+                else:
+                    is_read_poll[poll] = True
+            else:
+                is_read_poll[poll] = poll.id not in last_modifieds
```

Comment:
```suggestion
            is_read_poll[poll] = poll.id not in last_modifieds or (poll.id in last_views and last_views[poll.id] > last_modifieds[poll.id])
```
Nie wiem, czy to poważna sugestia. Ale można przynajmniej tak zmienić kolejność gałęzi, by się mniej zagnieżdżać.